README.md

```diff
@@ -216,3 +218,55 @@ const schema = Enjoi.schema({
     ]
 });
 ```
+
+### Extend Joi validation
+
+Example with root schema:
+```javascript
+const schema = Enjoi.schema({
+    type: 'object',
+    properties: {
+        title: {type: 'string'},
+    },
+    definitions: {
+        Name: {
+            type: 'string'
+        }
+    }
+}, {
+    extendValidation: {
+        '#': Joi.object().keys({title: Joi.string().max(3)}),
+    }
+});
+```
+
+Example with root schema and references:
+```javascript
+const schema = Enjoi.schema({
+    type: 'object',
+    properties: {
+        title: {type: 'string'},
+        name: {$ref: '#/definitions/Name'},
+        address: {$ref: 'definitions#/Address'}
+    },
+    definitions: {
+        Name: {
+            type: 'string'
+        }
+    }
+}, {
+    subSchemas: {
+        'definitions': {
+            'Address': {
+                'type': 'string'
+            }
+        }
+    },
+    extendValidation: {
+        '#': Joi.object().keys({title: Joi.string().max(3)}),
+        '#/definitions/Name': Joi.string().max(8),
```

Comment:
Hard to use this as an example to illustrate why this is useful. Could use currently supported json-schema to specify.

Reply:
You are right. What do you think about this example:
```
const schema = Enjoi.schema({
    type: 'object',
    required: ['min', 'max'],
    properties: {
        lower: {type: 'number', minimum: 5},
        higher: {type: 'number', maximum: 20},
    },
}, {
    extendValidation: {
        '#': Joi.object().keys({
            higher: Joi.number().greater(Joi.ref('lower')),
        }),
    },
});
```